torch/__init__.py

```diff
@@ -1752,6 +1752,90 @@ def registerOp(cls, op_key, full_schema, op_impl, dispatch_key):
+        - PyTree leaf types (such as torch.Tensor, int, float, bool)
+        - Data structures that can be flattened by PyTree
+            (e.g., dict, list, tuple, namedtuple, OrderedDict)
+        - Dataclass subclass (must be registered with 
+            torch._export.utils.register_dataclass_as_pytree_node` first)
+
+    torch.export() specializes the traced program based on the values of 
+    inputs that are not torch.Tensors. If you want to preserve dynamic
+    branching behavior in the generated graph, you will need to manually
+    rewrite the source code using control flow operations like
+    `torch.ops.higher_order.cond`.
+
+    While tracing, torch.export() takes note of shape-related assumptions 
+    made by the user program and the underlying PyTorch operator kernels.
+    The output ExportedProgram is considered valid only when these
+    assumptions hold true.
+
+    The returned ExportedProgram maintains the following invariants:
+        - It is guaranteed to be a faithful representation of the original
+            program.
+        - It maintains the exact calling convention of the original program.
+        - It contains a `state_dict` that stores the `torch.nn.Parameters`
+            of the original program.
+        - It includes an fx.GraphModule that represents the computation of
+            the original program. The GraphModule:
+            - Contains only `placeholder`, `call_function`, and `return` nodes.
+            - Inlines all submodules from the original programs.
+            - Lifts all parameters and buffers of the original program as
+                inputs to the graph.
+            - Does not mutate intermediate values, parameters, or buffers.
+            - Does not include operations with side effects.
+            - Contains only a curated subset of ATen operations and registered
+                custom operations (by default).
+                See the list of Core ATen Ops here: https://pytorch.org/docs/stable/ir.html
+
+    Args:
+        f: The `nn.Module` or callable to trace. If an `nn.Module` is provided,
+             its `forward` method will be traced.
+        args: Example positional inputs.
+        kwargs: Optional example keyword inputs.
+        constraints: An optional list of constraints on the dynamic arguments
+            that specify their possible range of shapes. By default, shapes of
+            input torch.Tensors are assumed to be static. If an input torch.Tensor
+            is expected to have dynamic shapes, please use `torch._export.dynamic_dim()`
+            to define `Constraint` objects that specify the dynamics and the possible
+            range of shapes.
+
+    Returns:
+        An ExportedProgram containing the traced callable.
+
+    For more information on how to analyze, transform, execute, and further lower an ExportedProgram,
+    please refer to the docstring of `torch._export.exported_program.ExportedProgram`.
```

Comment:
@thiagocrepaldi Would be nice to mention here that this tracing process can be done under fake mode, so you don't actually run any compute

Reply:
I believe Ed mentioned the use case in which users can

```python
with torch._subclasses.FakeTensorMode():
    fake_mode = create_model()
    fake_args, fake_kwargs = create_input()
    torch.export(fake_model)(*fake_args, **fake_kwargs)
```

to ensure that all model and input are 100% fake. This is a pattern used by the new ONNX exporter to properly export a model that cannot fit in memory, preventing OOM in the process.